lib/serializer.js

```diff
@@ -379,7 +379,25 @@ function handleIncludes (resp, includes, relations, app) {
   })
 
   if (embedded.length) {
-    resp.included = _.flattenDeep(embedded)
+    // This array may contain duplicate models if the same item is referenced multiple times in 'data'
+    let duplicate = _.flattenDeep(embedded)
+    // So begin with an empty array that will only contain unique items
+    let unique = []
+    // Iterate through each item in the first array
+    duplicate.forEach(function(d){
+      // Count the number of items in the unique array with matching 'type' AND 'id'
+      // Since we're adhering to the JSONAPI spec, both 'type' and 'id' can assumed to be present
+      // Both 'type' and 'id' are needed for comparison because we could theoretically have objects of different
+      // types who happen to have the same 'id', and those would not be considered duplicates
+      let count = unique.filter(function(u){
+        return (u.type === d.type) && (u.id === d.id)
+      }).length;
+      // If there are no matching entries, then add the item to the unique array
+      if (count === 0) {
+        unique.push(d)
+      }
+    })
```

Comment:
We have the lodash library available for this project so you could use the `uniqBy` or `uniqWith` functions like so:

```js
// uniqWith
unique = _.uniqWith(duplicate, function (a, b) { return a.type === b.type && a.id === b.id })

// uniqBy
unique = _.uniqBy(duplicate, function (obj) { return obj.id + obj.id })
```